### PR TITLE
Build all iOS example apps on current Flutter stable

### PR DESCRIPTION
**Pull request by jmagman**
Run `build-examples --ipa` on Flutter 2.2.2 (current stable).  Allow the Flutter tool and CocoaPods to migrate project files.

This will reduce working copy changes when the example projects are built for testing.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by jmagman** [comment]
Should I update the `CHANGELOG`s with `## NEXT`s?

File: packages/battery/battery/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -9,10 +9,6 @@
 /* Begin PBXBuildFile section */
 		1498D2341E8E89220040F4C2 /* GeneratedPluginRegistrant.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */; };
 		3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };
-		3B80C3941E831B6300D905FE /* App.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 3B80C3931E831B6300D905FE /* App.framework */; };
```
Comment (jmagman) [inline packages/battery/battery/example/ios/Runner.xcodeproj/project.pbxproj:12]: flutter/flutter#51453

File: packages/battery/battery/example/ios/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -315,7 +287,6 @@
 /* Begin XCBuildConfiguration section */
 		97C147031CF9000F007C117D /* Debug */ = {
 			isa = XCBuildConfiguration;
-			baseConfigurationReference = 9740EEB21CF90195004384FC /* Debug.xcconfig */;
```
Comment (jmagman) [inline packages/battery/battery/example/ios/Runner.xcodeproj/project.pbxproj:318]: flutter/flutter#54691

File: packages/battery/battery/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 3]
```diff
@@ -2,6 +2,6 @@
 <Workspace
    version = "1.0">
    <FileRef
-      location = "group:Runner.xcodeproj">
```
Comment (jmagman) [inline packages/battery/battery/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata:5]: flutter/flutter#73508

File: packages/connectivity/connectivity/example/ios/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -229,7 +216,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" thin";
+			shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" embed_and_thin";
```
Comment (jmagman) [inline packages/connectivity/connectivity/example/ios/Runner.xcodeproj/project.pbxproj:219]: flutter/flutter#51453

File: packages/path_provider/path_provider/example/linux/flutter/generated_plugin_registrant.cc [hunk 5]
```diff
@@ -2,8 +2,6 @@
 //  Generated file. Do not edit.
 //
 
-// clang-format off
```
Comment (jmagman) [inline packages/path_provider/path_provider/example/linux/flutter/generated_plugin_registrant.cc:5]: flutter/flutter#76662

File: packages/battery/battery/example/ios/Runner.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -229,22 +216,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" thin";
-		};
-		4096151B6BA12D6D4D7DD96A /* [CP] Embed Pods Frameworks */ = {
```
Comment (jmagman) [inline packages/battery/battery/example/ios/Runner.xcodeproj/project.pbxproj:234]: The Objective-C Podfiles are set up to use libraries, not frameworks.  CocoaPods is removing the build step that embeds the frameworks into the app.

**Comment by jmagman** [comment]
Hm, the formatter doesn't like the `generated_plugin_registrant`...
```diff
-
-void RegisterPlugins(flutter::PluginRegistry* registry) {
-}
+void RegisterPlugins(flutter::PluginRegistry* registry) {}
```

**Comment by jmagman** [comment]
> Hm, the formatter doesn't like the generated empty `generated_plugin_registrant`...

I reverted the Windows and Linux project changes, sticking to just iOS for this PR.

**Comment by stuartmorgan** [comment]
> Hm, the formatter doesn't like the generated empty `generated_plugin_registrant`...
> 
> ```diff
> -
> -void RegisterPlugins(flutter::PluginRegistry* registry) {
> -}
> +void RegisterPlugins(flutter::PluginRegistry* registry) {}
> ```

Yeah, this is why we turned off auto-format in the templates recently on master.

**Review by stuartmorgan — APPROVED** [review]
RSLGTM

**Pull request merged**